client/components/sites-dropdown/index.jsx

```diff
@@ -81,7 +67,7 @@ export default class SitesDropdown extends PureComponent {
 						{
 							this.props.isPlaceholder
 							? <SitePlaceholder />
-							: <Site site={ this.getSelectedSite() } indicator={ false } />
+							: <Site site={ this.props.selectedSite } indicator={ false } />
```

Comment:
Hmm, I think we'll have to use `this.state.selectedSiteId` here. Meaning we also need a bound `getSite()` passed from `connect()`.